src/metabase/api/collection.clj

```diff
@@ -317,22 +327,26 @@
   and queries that are technically executable without parameters can be unacceptably slow
   without the necessary constraints. (Marking parameters in optional blocks as required doesn't
   seem to be useful any way, but if the user said it is required, we honor this flag.)"
-  [row]
-  (let [native-query (-> row :dataset_query (json/parse-string keyword) :native)]
+  [text template-tags]
+  (let [obligatory-params (into #{}
+                                (comp (filter params/Param?)
+                                      (map :k))
+                                (params.parse/parse text))]
+    (and (every? #(or (:default %) (= (:type %) :dimension) (fully-parametrized-snippet? %))
```

Comment:
```suggestion
    (and (every? #(or (:default %) 
                      (= (:type %) :dimension) 
                      (fully-parametrized-snippet? %))
```

super nitty but It's hard for me to read an anonymous function with too many parentheses in a row.